### DISCUSSION TITLE
Ports with similar names

### DISCUSSION THREAD

**Discussion by DAT4** [plantuml/plantuml#1325] [Q&A] [answered]
Hey I am trying to make ports with similar names on different components. fx.

```plantuml
node srv1 {
 portin br0
}

node srv2 {
 portin br0
}
```

I tried with 

```plantuml
srv1.br0 -> srv2.br0
```

and 

```plantuml
node srv2 {
 portin "br0" srv2br0
}
```

But it didn't work.

**Comment by The-Lum** [comment, marked as answer]
Hi @DAT4, and all,

You can use `alias` as:

```puml
node srv1 {
  portin "br0" as srv1_br0
}
node srv2 {
  portin "br0" as srv2_br0 
}

srv1_br0 -> srv2_br0
```

_If that can help._
Regards.

**Reply by DAT4** [reply]
Oh yes - I forgot "as" :):):))

**Comment by arnaudroques** [comment]
However, there is something that sounds like a bug.
The [following example](http://www.plantuml.com/plantuml/umla/SoWkIImgAStDuIfEBL4eJYr8B4eioIzIqEFYoijFILKeBYepLAZcKYZ8Boh9p5D8AZBWgaL86M5AWJJe0TaAkhfsO2KW3fT3QbuAq180) *should* work:

```
@startuml
set separator .

node srv1 {
 portin br0
}

node srv2 {
 portin br0
}

srv1.br0 --> srv2.br0
@enduml
```

**Comment by evantill** [comment]
actual result on [PlantUML Online Server](https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuIfEBL4eJYr8B4eioIzIqEFYoijFILKeBYepLAZcKYZ8Boh9p5D8AZBWgaL86M5AWJJe0TaAkhfsO2KW3fT3QbuAq180)

![image](https://user-images.githubusercontent.com/609288/224961853-c9462e47-c68e-4ec0-ad27-dbca5fb741a9.png)

**Reply by evantill** [reply]
and the [result using aliases](https://www.plantuml.com/plantuml/uml/SoWkIImgAStDuIfEBL4eJYr8B4eioIzIqEFYoijFILKeBYepLAZcKYZ8Boh9p5D8AZBWgaL86M5AWJJe0TaAkhfsO2KW3fT3QbuAq180)

![image](https://user-images.githubusercontent.com/609288/224962340-6a83ea99-5972-4ca6-8d6a-e48e9f4214c3.png)

**Reply by arnaudroques** [reply]
This is the expected result, even when you don't use aliases. This will be fixed soon.

**Reply by evantill** [reply]
do you need an issue for tracing that ?

**Reply by arnaudroques** [reply]
No, no need for that. We will use this discussion as an issue tracing
(except if you find this too dirty :-) )

**Comment by arnaudroques** [comment]
This is fixed [in last snapshot](https://www.plantuml.com/plantuml/umla/SoWkIImgAStDuIfEBL4eJYr8B4eioIzIqEFYoijFILKeBYepLAZcKYZ8Boh9p5D8AZBWgaL86M5AWJJe0TaAkdPW5I0sbqDgNWhG3W00).
Thanks!
